Classes/Service/ContentService.php

```diff
@@ -345,24 +345,38 @@ protected function initializeRecordByNewAndOldAndLanguageUids(
                 // look for the translated version of the parent record indicated
                 // in this new, translated record. Below, we adjust the parent UID
                 // so it has the UID of the translated parent if one exists.
-                $translatedParents = (array) $this->workspacesAwareRecordService->get(
-                    'tt_content',
-                    'uid,sys_language_uid',
-                    "t3_origuid = '" . $oldRecord['tx_flux_parent'] . "'" . BackendUtility::deleteClause('tt_content')
-                );
-                foreach ($translatedParents as $translatedParent) {
-                    if ($translatedParent['sys_language_uid'] == $newLanguageUid) {
-                        // set $translatedParent to the right language ($newLanguageUid):
-                        break;
+                unset($copiedParent);
+                if (0 < (integer) $oldRecord['tx_flux_parent']) {
+                    // Search for a container element which has been copied from
+                    // the container referred to by the original element
+                    // (oldRecord) and exists in the target language.
+                    // Note that this cannot guarantee the correct container
+                    // to be found and used (using copies for translations
+                    // leads to ambiguous results by principle).
+                    // This is the old method which used to implement a fix for
+                    // #816 and #796.
+                    $copiedParents = (array) $this->workspacesAwareRecordService->get(
+                        'tt_content',
+                        'uid,sys_language_uid',
+                        "t3_origuid = '" . (integer) $oldRecord['tx_flux_parent'] . "' AND " .
+                            "pid = '" . (integer) $row['pid'] . "' AND " .
+                            "CType = 'fluidcontent_content' AND " .
```

Comment:
Thanks for the explanation. :)

I now agree we should remove the CType check introduced in 9ffecce064 and re-formatted in db9177ea. With that commit, I also introduced a condition for the container to be on the same page - that one should stay in as a sanity check since the "translated" container of a to-be-translated record should be on the same page or is there something else I missed?